game.py

```diff
@@ -20,22 +22,43 @@ def snowman(snowman_word):
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    wrong_guesses_list = []
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
 
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES and not is_word_guessed(snowman_word, correct_letter_guess_statuses):
+
+        print_word_progress_string(snowman_word, correct_letter_guess_statuses)  # Show word progress
+        user_input = get_letter_from_user(correct_letter_guess_statuses,wrong_guesses_list)
+
+        if user_input in snowman_word:
+            for letter in snowman_word:
+                if letter == user_input:
+                    correct_letter_guess_statuses[letter] = True
```

Comment:
I fully get where you are coming from with this particular approach! However, remember that the function on line 26 creates a dictionary called `correct_letter_guess_statuses`. With a dictionary, we don't need to loop through the snowman word to find the letter we are looking for. We can get away with just accessing the letter we've input directly and changing its status to True!

```suggestion
          correct_letter_guess_statuses[user_input] = True
```